sway-fmt-v2/src/utils/comments.rs

```diff
@@ -14,7 +14,12 @@ pub struct CommentSpan {
 
 impl Ord for CommentSpan {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.start.cmp(&other.start)
+        // If the starting position is the same encapsulatig span (i.e, wider one) should come
+        // first
+        match self.start.cmp(&other.start) {
+            Ordering::Equal => self.end.cmp(&other.end),
```

Comment:
```suggestion
            Ordering::Equal => other.end.cmp(&self.end),
```
^ I *think* it should be this